direnv.rc

```diff
@@ -34,6 +34,6 @@ function use_flox() {
     direnv_load flox activate "$@" -- "$direnv" dump
 
     if [[ $# == 0 ]]; then
-        watch_dir ".flox"
+        watch_file ".flox/env.json"
```

Comment:
> If we watch .flox/env.json I believe that should change every time the environment meaningfully changes?

It looks like this only changes if the environment's name does. The version is for the format of the file rather than the contents of the environment. So we wouldn't pick up changes to the manifest such as `hook` and `vars` (but `install` is unaffected because it's already on the `PATH`).

Do you have an example environment for the issue that you were seeing? Does it populate `.flox/cache` as a local store?

I can't see support for excluding dirs or patterns in `watch_dir` but we could combine `watch_file .flox/env.json` and `watch_dir .flox/env`. I don't think that changes to `.flox/run` should matter since they would coincide with a change to the lockfile and/or activation.

Reply:
I can't locally reproduce the same problem you saw where every `ls` caused a re-activation after:
1. Checkout https://github.com/flox/floxenvs/pull/17
2. `cd envs/direnv && flox activate`
3. `cd ../flaim && floxit`

I can see that `PYTHON_DIR="$FLOX_ENV_CACHE/python"` ([here](https://github.com/flox/floxenvs/blob/e4ce186d67db704e99329cf95bdbcfdd0158a79a/flaim/manifest.toml#L24)) though, which would cause more re-activations than necessary or expected, so we certainly want to exclude `.flox/cache`.

Watching `.flox/env/manifest.lock` and not `.flox/env/manifest.toml` seems correct because `flox activate` will only pick up manifest changes from the lock and that won't happen until you `flox edit [--sync]`.

There is also a `.flox/env.lock` for "managed environments" that contains local and remote revs, which is probably what you were thinking of previously. I'm not quite familiar enough with the workflow for managed envs but it seems like it would be good to also watch that file.

Reply:
> Watching .flox/env/manifest.lock and not .flox/env/manifest.toml seems correct because flox activate will only pick up manifest changes from the lock and that won't happen until you flox edit [--sync].

activate should relock/rebuild if the manifest changed even without a flox edit


env.lock changes with every push or imperative change to managed environments

lastly  (for path envs) .flox/run/* will be updated when you build an environment

Reply:
If you want direnv to pick up changes before they are synched or in path envy’s before they are built, we should watch the manifest.toml file rather than the lock as the lock

Reply:
env.json only changes when we change the name of a path env or push an environment

Reply:
@dcarley Indeed, I was thinking of env.lock. But yes I think it only is useful for tracking change in remote envs.

I think I will try `.flox/env/` and `.flox/env.json` for now and see how it goes.